prime-angular-frontend/src/app/shared/models/banner.model.ts

```diff
@@ -1,9 +1,61 @@
+import { Moment } from 'moment';
 import { BannerType } from '@shared/enums/banner-type.enum';
+import moment from 'moment';
+// import * as moment from 'moment';
 
 export interface Banner {
   bannerType: BannerType;
   title: string;
   content: string; // HTML markup
   startDate: string;
+  startTime: string;
   endDate: string;
+  endTime: string;
+}
+
+export class BannerViewModel {
+  public bannerType: BannerType;
+  public title: string;
+  public content: string;
+  public startDate: Moment;
+  public endDate: Moment;
+
+  public static toBanner(bannerVm: BannerViewModel): Banner {
+    const start = moment(bannerVm.startDate).local();
+    const end = moment(bannerVm.endDate).local();
+
+    return {
+      bannerType: bannerVm.bannerType,
+      title: bannerVm.title,
+      content: bannerVm.content,
+      startDate: start.format(),
+      startTime: start.format("HHmm"),
+      endDate: end.format(),
+      endTime: end.format("HHmm")
+    };
+  }
+
+  public static fromBanner(banner: Banner): BannerViewModel {
+    const startDate = moment(banner.startDate);
+    const startTime = moment(banner.startTime, 'HHmm');
+    const endDate = moment(banner.endDate);
+    const endTime = moment(banner.endTime, 'HHmm');
+
+    const start = startDate.set({
+      hour: startTime.get('hour'),
+      minute: startTime.get('minute')
+    });
+    const end = endDate.set({
+      hour: endTime.get('hour'),
+      minute: endTime.get('minute')
+    });
+
+    const vm = new BannerViewModel();
```

Comment:
This is why I tend to add a constructor so other people don't have to type out the assignments each time they new-up an object, but that's just personal preference since I find assignments like this to be tedious when replicated and can instead use destructuring to pass as params in 2 lines.  In this case what are the chances anyone else is going to be populate a `BannerViewModel` with these keys outside of this factory method... really low... so completely up to you there's no wrong answer.